apps/platform/trpc/routes/branch.ts

```diff
@@ -158,4 +162,53 @@ export const branches = createRouter({
 
       return updatedBranch;
     }),
+  delete: withAuth
+    .input(z.object({ branchId: z.string() }))
+    .mutation(async ({ ctx, input }) => {
+      const { branchId } = input;
+
+      try {
+        const softDeletedBranch = await ctx.prisma.branch.update({
+          where: {
+            id: branchId,
+          },
+          data: {
+            deletedAt: new Date(),
+          },
+        });
+
+        return softDeletedBranch;
+      } catch (error) {
+        throw new TRPCError({
+          code: "BAD_REQUEST",
+          message: "Branch does not exist",
+        });
+      }
+    }),
+  restoreBranch: withAuth
+    .input(
+      z.object({
+        branchId: z.string(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      const { branchId } = input;
```

Comment:
Before `update` we should check If the branch is actually deleted or not. Just to be on the safe side.